server/analysis/gp.py

```diff
@@ -20,8 +20,7 @@ class GPRNP(object):
     MAX_TRAIN_SIZE = 7000
     BATCH_SIZE = 3000
```

Comment:
Could you please remove these parameters and pass them at the asynchronous task. You may also want to add the two parameters in the _init_() function.

Reply:
MAX_TRAIN_SIZE is currently used as a class data member in check_X_y, which is a static method.  So if we only pass in MAX_TRAIN_SIZE through __init__, and save it as an instance variable like all the others, then I'm not sure how check_X_y can know MAX_TRAIN_SIZE.  Advice?

Reply:
Can we change the static method  check_X_y to a normal function (i.e. remove the static method decorator) ? What do you think ?

Reply:
Yes, that would work.  I'm not familiar enough with Python internals to know whether removing staticmethod has any performance impacts or is purely for visual convenience, but if you're good with it then I am. Will fix after today's meeting.